src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.CoreCLR.cs

```diff
@@ -11,7 +11,15 @@ public abstract partial class EqualityComparer<T> : IEqualityComparer, IEquality
     {
         // To minimize generic instantiation overhead of creating the comparer per type, we keep the generic portion of the code as small
         // as possible and define most of the creation logic in a non-generic class.
-        public static EqualityComparer<T> Default { [Intrinsic] get; } = (EqualityComparer<T>)ComparerHelpers.CreateDefaultEqualityComparer(typeof(T));
+        public static EqualityComparer<T> Default
+        {
+            // NoInlining helps JIT to track get_Default() accross all phases.
+            [Intrinsic, MethodImpl(MethodImplOptions.NoInlining)]
+            get
+            {
+                return (EqualityComparer<T>)ComparerHelpers.CreateDefaultEqualityComparer(typeof(T));
```

Comment:
Is this going to cause compared allocation every time this is called when the optimizations are off? That's too expensive even with optimizations off.

Reply:
@jkotas  I can disable inlining for this call in jit (when optimizations are enabled) - btw, that's how I collected jit-diffs.